### PR TITLE
Update Android Gradle Plugin to version 7.0.2

### PR DESCRIPTION
**Pull request by litetex**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Updated ``com.android.tools.build:gradle`` to 7.0.1 (recommended through Android Studio)
- Release notes are available at: https://developer.android.com/studio/releases/gradle-plugin#7-0-0

#### APK testing 
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by litetex** [comment]
Update: The build requires now Java 11 so the GH-action workflows have to be adjusted

File: build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.2.2'
+        classpath 'com.android.tools.build:gradle:7.0.1'
```
Comment (TobiGr) [inline build.gradle:11]: I am asked to updated to `7.0.2`

**Review by TobiGr — COMMENTED** [review]
Looks good to me.

File: build.gradle (hunk 1 above)
Comment (litetex) [reply build.gradle]: Okay weird sentence....
I updated to 7.0.2 (although I don't find any changelogs for that version at https://developer.android.com/studio/releases/gradle-plugin#7-0-0).

Should be fine now 😄

Comment (TobiGr) [reply build.gradle]: asked by Android Studio :sweat_smile:

**Review by Stypox — APPROVED** [review]
I think we can switch to building with Java 11 without problems. I already build with JDK 11

**Review by TobiGr — APPROVED** [review]

**Comment by opusforlife2** [comment]
> I did not find a changelog, too.

https://androidstudio.googleblog.com/2021/09/android-studio-arctic-fox-202031-patch.html

@litetex @TobiGr This, probably. Two issues under AGP.

**Comment by Atemu** [comment]
This breaks compat with pure IDEA without google's ~~telemetry~~ android flavour.

See also https://youtrack.jetbrains.com/issue/IDEA-281657

Idea users need to revert this when they want to do anything.  
The AGP 4.2.2 version works just fine FYI, would it be acceptable to revert this until Idea supports AGP 7.0?

**Comment by litetex** [comment]
> This breaks compat with pure IDEA without google's ~telemetry~ android flavour.
> 
> See also https://youtrack.jetbrains.com/issue/IDEA-281657
> 
> Idea users need to revert this when they want to do anything. The AGP 4.2.2 version works just fine FYI, would it be acceptable to revert this until Idea supports AGP 7.0?

This is fixed/resolved: https://youtrack.jetbrains.com/issue/IDEA-275441#focus=Comments-27-5223246.0-0

**Comment by Atemu** [comment]
Only in the early access builds though.

It'd just be nice to be a bit more conservative with major revision bumps like these in the future then.

**Pull request merged**